collector/processes_linux.go

```diff
@@ -111,7 +113,7 @@ func (c *processCollector) getAllocatedThreads() (int, map[string]int32, int, er
 	for _, pid := range p {
 		stat, err := pid.Stat()
 		// PIDs can vanish between getting the list and getting stats.
-		if errors.Is(err, os.ErrNotExist) {
+		if errors.Is(err, os.ErrNotExist) || errorContains(err, syscall.ESRCH) {
```

Comment:
Hrm, let's refactor this a bit.

```suggestion
		if err != nil {
			if errors.Is(err, os.ErrNotExist) || strings.Contains(err.Error(), syscall.ESRCH.Error()) {
```

This eliminates the need for a helper function, and having a bunch of extra `nil` tests in that helper.

Reply:
👍